lib/ansible/plugins/inventory/foreman.py

```diff
@@ -121,134 +126,139 @@ def _get_session(self):
             self.session.verify = self.get_option('validate_certs')
         return self.session
 
-    def _get_json(self, url, ignore_errors=None):
-
-        if not self.use_cache or url not in self._cache.get(self.cache_key, {}):
+    def _get_all_hosts(self, url):
 
-            if self.cache_key not in self._cache:
+        if self.use_cache:
+            try:
+                inventory = self._cache[self.cache_key][url]
+            except KeyError:
+                # The cache key expired or does not exist yet
+                inventory = None
                 self._cache[self.cache_key] = {url: ''}
```

Comment:
And this `self._cache.set(self.cache_key, {url: ''})`

Reply:
This is also correct because the `self._cache` interface is dict like. The old set() method works for backwards compatibility for the former `self.cache` interface, but is deprecated and will be removed in 2.12. Here are also developer docs for inventory caching https://docs.ansible.com/ansible/latest/dev_guide/developing_inventory.html#inventory-cache.